src/sentry/utils/snuba.py

```diff
@@ -29,6 +29,18 @@
 MAX_HASHES = 5000
 
 SENTRY_SNUBA_MAP = {
+    # general
+    'event_id': 'event_id',
+    'project_id': 'project_id',
+    'message': 'message',
+    'issue': 'issue',
+    'group_id': 'group_id',
+    'timestamp': 'timestamp',
+    'time': 'time',
+    'deleted': 'deleted',
```

Comment:
This column should never be used by clients. All snuba queries will filter out deleted rows internally anyway

Reply:
Same with
retention_days: used internally to figure out when to delete old data
group_id: clients should always use `issue` and not query `group_id` directly.
received: This is also for internal use, I don't think we have ever exposed this to customers?

Reply:
I'm not sure that this is the place to restrict that, I guess removing it from this list would just mean that queries for `deleted` would become `tags[deleted]` which is not what we want either.

Reply:
I don't think any of this is going to be exposed intentionally, but it's always possible for someone to call the API with this directly or via search box input in event stream. If the terms group_id, deleted, retention_days are never communicated or prompted in anyway, is it safe to assume the someone who has intentionally input these terms is probably looking for a custom tag with that name that they've created?

Reply:
yeah that makes sense. I guess the best we can do is turn `"deleted"` into `tags[deleted]` to see if its a custom tag for that user.

Actually that makes me wonder. if someone creates a custom tag with the same name as one of our columns, is there any way for them to query for it in discover? IDK if that's a priority or anything, just wondering.

Reply:
We could add support for the old syntax if that's a priority

Reply:
nah, I don't mind, was just wondering.